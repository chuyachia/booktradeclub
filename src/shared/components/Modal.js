import Alert from "./Alert";
import {addBook,closeBook,removeBook} from "../actions/bookAction";
import {connect} from "react-redux";
import {addRequest} from "../actions/requestAction";
import {getUsersLocation} from "../actions/userAction";
import React from 'react';
import ReactModal from 'react-modal';
import {Redirect } from "react-router-dom";


class Modal extends React.Component{
    constructor(props){
        super(props);
    }
    closeBook(){
        this.props.dispatch(closeBook());
    }
    addBook(){
        this.props.dispatch(addBook(this.props.info,this.props.username));
    }
    removeBook(){
        this.props.dispatch(removeBook(this.props.info.bookId,this.props.username));
    }
    addRequest(indx){
        var receiver = this.props.info.ownBy[indx];
        this.props.dispatch(addRequest(this.props.username,receiver,this.props.info.bookId,this.props.info.title,this.props.email));
    }
    componentWillReceiveProps(nextProps){
        if (nextProps.btnuse=="addrequest"&&nextProps.info.ownBy&&nextProps.info!=this.props.info){
            this.props.dispatch(getUsersLocation(nextProps.info.ownBy));
        }
    }
    render(){
        var modalStyles = {overlay: {zIndex: 10}};
        if (this.props.tologin){
            return(<Redirect to={{
            pathname: "/connect"
          }}/>);
        } else {
            var ownedbooks = this.props.ownedbooks.map(book=>book.bookId);
            var outrequests = this.props.outrequests.filter(request=>request.receiver.bookId==this.props.info.bookId&&request.status!="declined");
            var outrequser = outrequests.map(request=>request.receiver.username);
            return(
            <ReactModal style={modalStyles}
               isOpen={this.props.open}
               ariaHideApp={false}
               contentLabel="Review Modal">
              <a class="text-dark" style={{float:"right",cursor:"pointer"}}><i class="fas fa-times" onClick={this.closeBook.bind(this)}/></a>
               <Alert/>
               <h3>{this.props.info.title}</h3>
               <dl>
                    <dt><img src={this.props.info.imageUrl} alt={this.props.info.title} class="img-thumbnail"/></dt>
               {this.props.info.authors&&
               (<div>
                    <dt>Author</dt>
                    <dd>{this.props.info.authors.map((author,i)=> {
                    if (i<this.props.info.authors.length-1) {
                        return (<span key={i}>{author}, </span>)
                    } else {
                        return (<span key={i}>{author}</span>)
                    }
                    })}</dd>
                </div>)}
                    <dt>Publisher</dt>
                    <dd>{this.props.info.publisher}</dd>
                    <dt>Published</dt>
                    <dd>{this.props.info.publishedDate}</dd>

                    <dd>{this.props.info.description}</dd>
               </dl>
               
               {this.props.btnuse=="addrequest"&&this.props.ownerslocation&&
               <div>
               <table class="table">
                  <thead>
                    <tr>
                      <th scope="col">Owner</th>
                      <th scope="col">Location</th>
                      <th scope="col"></th>
                    </tr>
                  </thead>
                  <tbody>
                    {this.props.info.ownBy.map((user,i)=>(
                    <tr key={i}>
                      <td>{user}</td>
                      <td>{this.props.ownerslocation[i]}</td>
                      <td>
                      {user!==this.props.username
                        ?outrequser.indexOf(user)==-1?
                          (<button class="btn btn-raised bg-dark text-light" onClick={()=>{this.addRequest(i)}}>Request</button>):
                          (<button class="btn" disabled>Already requested</button>)
                          :null
                      }</td>
                    </tr>
                    ))}
                  </tbody>
                </table></div>}
                
               {this.props.btnuse=="addbook"&&(ownedbooks.indexOf(this.props.info.bookId)==-1
                    ?<button class="btn btn-raised bg-dark text-light" onClick={this.addBook.bind(this)}>Add to my books</button>
                    :<button class="btn" disabled>Already added</button>)
               }
               
               {this.props.btnuse =="answersender"&&
               (<button class="btn btn-raised bg-dark text-light" onClick={()=>this.props.addexchange(this.props.info.bookId,this.props.info.title)}>Exchange</button>)}
                
                {this.props.btnuse =="removebook"&&
               (<button class="btn btn-raised bg-dark text-light" onClick={this.removeBook.bind(this)}>Remove from my books</button>)}
                
                </ReactModal>
            );
        }
    }
}
var propsMap = (store)=>{
    return{
        btnuse: store.viewbook.btnuse,
        tologin:store.viewbook.tologin,
        open:store.viewbook.open,
        info:store.viewbook.info,
        username : store.userinfo.username,
        email:store.userinfo.email,
        ownedbooks:store.userinfo.ownedbooks,
        outrequests:store.userinfo.outrequests,
        ownerslocation:store.viewbook.ownerslocation
    };
};

export default connect(propsMap)(Modal);