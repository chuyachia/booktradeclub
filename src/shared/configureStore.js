import {applyMiddleware,createStore} from "redux";
import logger from "redux-logger";
import promise from "redux-promise-middleware";
import reducer from "./reducers";
import thunk from "redux-thunk";

const middleware = applyMiddleware(promise(),thunk,logger());

const configureStore = preloadedState =>
  createStore(reducer, preloadedState, middleware);
  
export default configureStore;