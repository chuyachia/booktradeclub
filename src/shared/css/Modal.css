.modal {
    position:fixed;
    width:100%;
    max-height:85%;
    overflow-y:scroll;
    top:50%;
    left:50%;
    transform: translate(-50%, -50%);
    z-index:15;
    visibility:hidden;
    opacity:0;
    transition:opacity 0.3s linear;
    background-color:white;
    padding: 10px;   
}


.modal.open, .overlay.open {
    visibility:visible;
    opacity:1;   
}
@media (min-width: 768px) {
    .modal {
        width:80%;
    }
}
@media (min-width: 992px) {
     .modal {
        width:60%;
    }   
}

.modal>i {
   float:right;
   cursor:pointer; 
   position:sticky;
   top:0px;
   text-align:right;
}
.overlay {
    background-color:rgb(204,204,204);
    background-color:rgba(204,204,204,0.3);
    position:fixed;
    width:100%;
    height:100%;
    top:0px;
    left:0px;
    z-index:10;
    visibility:hidden;
    opacity:0;
}
