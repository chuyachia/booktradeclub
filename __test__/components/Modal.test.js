import '../enzymeSetup';
import Modal from '../../src/shared/components/Modal';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import React from 'react';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';



const mockStore = configureStore();

describe('Modal Snapshot',()=>{
    let data;
    beforeEach(()=>{
        data = {
            viewbook:{
                btnuse: '',
                tologin:false,
                open:true,
                info:{
                    authors:[],
                    bookId:"",
                    categories:[],
                    description:"",
                    imageUrl:"",
                    ownBy:[],
                    publishedDate:"",
                    publisher:"",
                    title:""
                },
            ownerslocation:[]
            }, userinfo:{
                username : null,
                email:null,
                ownedbooks:[],
                outrequests:[]                
            }, alerts:{
                open:true,
                message:'test'
            }
        };
    });
    
    it('Snapshot of Modal for add request use', () => {
        data.viewbook.btnuse = 'addrequest';
        var store = mockStore(data);
        expect(EnzymeToJson(mount(<Provider store={store}><Modal/></Provider>))).toMatchSnapshot();
    });
    
    it('Snapshot of Modal for add book use', () => {
        data.viewbook.btnuse = 'addbook';
        var store = mockStore(data);
        expect(EnzymeToJson(mount(<Provider store={store}><Modal/></Provider>))).toMatchSnapshot();
    });

    it('Snapshot of Modal for answer sender use', () => {
        data.viewbook.btnuse = 'answersender';
        var store = mockStore(data);
        expect(EnzymeToJson(mount(<Provider store={store}><Modal/></Provider>))).toMatchSnapshot();
    });
    
    it('Snapshot of Modal for remove book use', () => {
        data.viewbook.btnuse = 'removebook';
        var store = mockStore(data);
        expect(EnzymeToJson(mount(<Provider store={store}><Modal/></Provider>))).toMatchSnapshot();
    });
});