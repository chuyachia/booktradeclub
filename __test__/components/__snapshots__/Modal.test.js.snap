// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Snapshot Snapshot of Modal for add book use 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Modal)>
    <Modal
      addBook={[Function]}
      addExchange={[Function]}
      addRequest={[Function]}
      btnuse="addbook"
      closeBook={[Function]}
      email={null}
      getUsersLocation={[Function]}
      info={
        Object {
          "authors": Array [],
          "bookId": "",
          "categories": Array [],
          "description": "",
          "imageUrl": "",
          "ownBy": Array [],
          "publishedDate": "",
          "publisher": "",
          "title": "",
        }
      }
      open={true}
      outrequests={Array []}
      ownedbooks={Array []}
      ownerslocation={Array []}
      removeBook={[Function]}
      tologin={false}
      username={null}
    >
      <Modal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="Review Modal"
        isOpen={true}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "overlay": Object {
              "zIndex": 10,
            },
          }
        }
      >
        <ModalPortal
          ariaHideApp={false}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="Review Modal"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "overlay": Object {
                "zIndex": 10,
              },
            }
          }
        >
          <div
            aria-modal="true"
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
                "zIndex": 10,
              }
            }
          >
            <div
              aria-label="Review Modal"
              className="ReactModal__Content ReactModal__Content--after-open"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": "4px",
                  "bottom": "40px",
                  "left": "40px",
                  "outline": "none",
                  "overflow": "auto",
                  "padding": "20px",
                  "position": "absolute",
                  "right": "40px",
                  "top": "40px",
                }
              }
              tabIndex="-1"
            >
              <i
                className="fas fa-times"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              />
              <Connect(Alert)>
                <Alert
                  closeAlert={[Function]}
                  open={true}
                >
                  <div>
                    <div
                      className="alert alert-warning alert-dismissible fade show"
                      role="alert"
                    >
                      <strong>
                        Oops! Something went wrong!
                      </strong>
                       The action just failed. Please try again later.
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="alert"
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                      </button>
                    </div>
                  </div>
                </Alert>
              </Connect(Alert)>
              <h3 />
              <dl>
                <dt>
                  <img
                    alt=""
                    className="img-thumbnail"
                    src=""
                  />
                </dt>
                <div>
                  <dt>
                    Author
                  </dt>
                  <dd />
                </div>
                <dt>
                  Publisher
                </dt>
                <dd />
                <dt>
                  Published
                </dt>
                <dd />
                <dd />
              </dl>
              <button
                className="btn btn-raised bg-dark text-light"
                onClick={[Function]}
              >
                Add to my books
              </button>
            </div>
          </div>
        </ModalPortal>
      </Modal>
    </Modal>
  </Connect(Modal)>
</Provider>
`;

exports[`Modal Snapshot Snapshot of Modal for add request use 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Modal)>
    <Modal
      addBook={[Function]}
      addExchange={[Function]}
      addRequest={[Function]}
      btnuse="addrequest"
      closeBook={[Function]}
      email={null}
      getUsersLocation={[Function]}
      info={
        Object {
          "authors": Array [],
          "bookId": "",
          "categories": Array [],
          "description": "",
          "imageUrl": "",
          "ownBy": Array [],
          "publishedDate": "",
          "publisher": "",
          "title": "",
        }
      }
      open={true}
      outrequests={Array []}
      ownedbooks={Array []}
      ownerslocation={Array []}
      removeBook={[Function]}
      tologin={false}
      username={null}
    >
      <Modal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="Review Modal"
        isOpen={true}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "overlay": Object {
              "zIndex": 10,
            },
          }
        }
      >
        <ModalPortal
          ariaHideApp={false}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="Review Modal"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "overlay": Object {
                "zIndex": 10,
              },
            }
          }
        >
          <div
            aria-modal="true"
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
                "zIndex": 10,
              }
            }
          >
            <div
              aria-label="Review Modal"
              className="ReactModal__Content ReactModal__Content--after-open"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": "4px",
                  "bottom": "40px",
                  "left": "40px",
                  "outline": "none",
                  "overflow": "auto",
                  "padding": "20px",
                  "position": "absolute",
                  "right": "40px",
                  "top": "40px",
                }
              }
              tabIndex="-1"
            >
              <i
                className="fas fa-times"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              />
              <Connect(Alert)>
                <Alert
                  closeAlert={[Function]}
                  open={true}
                >
                  <div>
                    <div
                      className="alert alert-warning alert-dismissible fade show"
                      role="alert"
                    >
                      <strong>
                        Oops! Something went wrong!
                      </strong>
                       The action just failed. Please try again later.
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="alert"
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                      </button>
                    </div>
                  </div>
                </Alert>
              </Connect(Alert)>
              <h3 />
              <dl>
                <dt>
                  <img
                    alt=""
                    className="img-thumbnail"
                    src=""
                  />
                </dt>
                <div>
                  <dt>
                    Author
                  </dt>
                  <dd />
                </div>
                <dt>
                  Publisher
                </dt>
                <dd />
                <dt>
                  Published
                </dt>
                <dd />
                <dd />
              </dl>
              <div>
                <table
                  className="table"
                >
                  <thead>
                    <tr>
                      <th
                        scope="col"
                      >
                        Owner
                      </th>
                      <th
                        scope="col"
                      >
                        Location
                      </th>
                      <th
                        scope="col"
                      />
                    </tr>
                  </thead>
                  <tbody />
                </table>
              </div>
            </div>
          </div>
        </ModalPortal>
      </Modal>
    </Modal>
  </Connect(Modal)>
</Provider>
`;

exports[`Modal Snapshot Snapshot of Modal for answer sender use 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Modal)>
    <Modal
      addBook={[Function]}
      addExchange={[Function]}
      addRequest={[Function]}
      btnuse="answersender"
      closeBook={[Function]}
      email={null}
      getUsersLocation={[Function]}
      info={
        Object {
          "authors": Array [],
          "bookId": "",
          "categories": Array [],
          "description": "",
          "imageUrl": "",
          "ownBy": Array [],
          "publishedDate": "",
          "publisher": "",
          "title": "",
        }
      }
      open={true}
      outrequests={Array []}
      ownedbooks={Array []}
      ownerslocation={Array []}
      removeBook={[Function]}
      tologin={false}
      username={null}
    >
      <Modal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="Review Modal"
        isOpen={true}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "overlay": Object {
              "zIndex": 10,
            },
          }
        }
      >
        <ModalPortal
          ariaHideApp={false}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="Review Modal"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "overlay": Object {
                "zIndex": 10,
              },
            }
          }
        >
          <div
            aria-modal="true"
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
                "zIndex": 10,
              }
            }
          >
            <div
              aria-label="Review Modal"
              className="ReactModal__Content ReactModal__Content--after-open"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": "4px",
                  "bottom": "40px",
                  "left": "40px",
                  "outline": "none",
                  "overflow": "auto",
                  "padding": "20px",
                  "position": "absolute",
                  "right": "40px",
                  "top": "40px",
                }
              }
              tabIndex="-1"
            >
              <i
                className="fas fa-times"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              />
              <Connect(Alert)>
                <Alert
                  closeAlert={[Function]}
                  open={true}
                >
                  <div>
                    <div
                      className="alert alert-warning alert-dismissible fade show"
                      role="alert"
                    >
                      <strong>
                        Oops! Something went wrong!
                      </strong>
                       The action just failed. Please try again later.
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="alert"
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                      </button>
                    </div>
                  </div>
                </Alert>
              </Connect(Alert)>
              <h3 />
              <dl>
                <dt>
                  <img
                    alt=""
                    className="img-thumbnail"
                    src=""
                  />
                </dt>
                <div>
                  <dt>
                    Author
                  </dt>
                  <dd />
                </div>
                <dt>
                  Publisher
                </dt>
                <dd />
                <dt>
                  Published
                </dt>
                <dd />
                <dd />
              </dl>
              <button
                className="btn btn-raised bg-dark text-light"
                onClick={[Function]}
              >
                Exchange
              </button>
            </div>
          </div>
        </ModalPortal>
      </Modal>
    </Modal>
  </Connect(Modal)>
</Provider>
`;

exports[`Modal Snapshot Snapshot of Modal for remove book use 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Modal)>
    <Modal
      addBook={[Function]}
      addExchange={[Function]}
      addRequest={[Function]}
      btnuse="removebook"
      closeBook={[Function]}
      email={null}
      getUsersLocation={[Function]}
      info={
        Object {
          "authors": Array [],
          "bookId": "",
          "categories": Array [],
          "description": "",
          "imageUrl": "",
          "ownBy": Array [],
          "publishedDate": "",
          "publisher": "",
          "title": "",
        }
      }
      open={true}
      outrequests={Array []}
      ownedbooks={Array []}
      ownerslocation={Array []}
      removeBook={[Function]}
      tologin={false}
      username={null}
    >
      <Modal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        closeTimeoutMS={0}
        contentLabel="Review Modal"
        isOpen={true}
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
        style={
          Object {
            "overlay": Object {
              "zIndex": 10,
            },
          }
        }
      >
        <ModalPortal
          ariaHideApp={false}
          bodyOpenClassName="ReactModal__Body--open"
          closeTimeoutMS={0}
          contentLabel="Review Modal"
          defaultStyles={
            Object {
              "content": Object {
                "WebkitOverflowScrolling": "touch",
                "background": "#fff",
                "border": "1px solid #ccc",
                "borderRadius": "4px",
                "bottom": "40px",
                "left": "40px",
                "outline": "none",
                "overflow": "auto",
                "padding": "20px",
                "position": "absolute",
                "right": "40px",
                "top": "40px",
              },
              "overlay": Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
              },
            }
          }
          isOpen={true}
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
          style={
            Object {
              "overlay": Object {
                "zIndex": 10,
              },
            }
          }
        >
          <div
            aria-modal="true"
            className="ReactModal__Overlay ReactModal__Overlay--after-open"
            onClick={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.75)",
                "bottom": 0,
                "left": 0,
                "position": "fixed",
                "right": 0,
                "top": 0,
                "zIndex": 10,
              }
            }
          >
            <div
              aria-label="Review Modal"
              className="ReactModal__Content ReactModal__Content--after-open"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "background": "#fff",
                  "border": "1px solid #ccc",
                  "borderRadius": "4px",
                  "bottom": "40px",
                  "left": "40px",
                  "outline": "none",
                  "overflow": "auto",
                  "padding": "20px",
                  "position": "absolute",
                  "right": "40px",
                  "top": "40px",
                }
              }
              tabIndex="-1"
            >
              <i
                className="fas fa-times"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                    "float": "right",
                  }
                }
              />
              <Connect(Alert)>
                <Alert
                  closeAlert={[Function]}
                  open={true}
                >
                  <div>
                    <div
                      className="alert alert-warning alert-dismissible fade show"
                      role="alert"
                    >
                      <strong>
                        Oops! Something went wrong!
                      </strong>
                       The action just failed. Please try again later.
                      <button
                        aria-label="Close"
                        className="close"
                        data-dismiss="alert"
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                      </button>
                    </div>
                  </div>
                </Alert>
              </Connect(Alert)>
              <h3 />
              <dl>
                <dt>
                  <img
                    alt=""
                    className="img-thumbnail"
                    src=""
                  />
                </dt>
                <div>
                  <dt>
                    Author
                  </dt>
                  <dd />
                </div>
                <dt>
                  Publisher
                </dt>
                <dd />
                <dt>
                  Published
                </dt>
                <dd />
                <dd />
              </dl>
              <button
                className="btn btn-raised bg-dark text-light"
                onClick={[Function]}
              >
                Remove from my books
              </button>
            </div>
          </div>
        </ModalPortal>
      </Modal>
    </Modal>
  </Connect(Modal)>
</Provider>
`;
